web/packages/teleterm/src/services/tshd/grpcContextBridgeClient.ts

```diff
@@ -0,0 +1,126 @@
+
+function validateAbortSignalType(abortSignal: AbortSignal) {
+  if (abortSignal instanceof AbortSignal) {
+    throw new Error(
+      'You must not pass AbortSignal instance. Use objectified version (tshAbortSignal).'
+    );
+  }
+}
+
+export function getObjectifiedInterceptors(): RpcInterceptor[] {
+  return [
+    {
+      interceptUnary(next, method, input, options) {
+        validateAbortSignalType(options.abort);
+
+        const output = next(method, input, options);
+        // UnaryCall is a class with a private field,
+        // so we can't assign an object to it.
+        // The workaround is satisfying only the public part of it
+        // and then casting to UnaryCall.
+        const objectifiedUnaryCall: PublicPart<UnaryCall> = {
+          ...output,
+          response: output.response.catch(e => objectifyError(e)),
+          then: (...args) =>
+            output.then(...args).catch(e => {
+              throw objectifyError(e);
+            }),
+        };
+        return objectifiedUnaryCall as UnaryCall;
+      },
+      interceptClientStreaming(next, method, options) {
+        validateAbortSignalType(options.abort);
+
+        const output = next(method, options);
+        const objectifiedClientStreamingCall: PublicPart<ClientStreamingCall> =
+          {
+            ...output,
+            then: (...args) =>
+              objectifiedClientStreamingCall.then(...args).catch(e => {
+                throw objectifyError(e);
+              }),
+            requests: objectifyRequests(output.requests),
+            response: output.response.catch(e => objectifyError(e)),
+          };
+        return objectifiedClientStreamingCall as ClientStreamingCall;
+      },
+      interceptServerStreaming(next, method, input, options) {
+        validateAbortSignalType(options.abort);
+
+        const output = next(method, input, options);
+        const objectifiedServerStreamingCall: PublicPart<ServerStreamingCall> =
+          {
+            ...output,
+            then: (...args) =>
+              objectifiedServerStreamingCall.then(...args).catch(e => {
+                throw objectifyError(e);
+              }),
+            responses: objectifyResponses(output.responses),
+          };
+        return objectifiedServerStreamingCall as ServerStreamingCall;
+      },
+      interceptDuplex(next, method, options) {
+        validateAbortSignalType(options.abort);
+
+        const output = next(method, options);
+        const duplexStreamingCall: PublicPart<DuplexStreamingCall> = {
+          ...output,
+          requests: objectifyRequests(output.requests),
+          responses: objectifyResponses(output.responses),
+          then: (...args) =>
+            duplexStreamingCall.then(...args).catch(e => {
+              throw objectifyError(e);
+            }),
+        };
+        return duplexStreamingCall as DuplexStreamingCall;
+      },
+    },
+  ];
+}
+
+export function objectifyClient<T>(client: ServiceInfo): T {
```

Comment:
I improved the type slightly. But I'm not sure if it is even possible to have a `ClassClient` type, since we expect a class instance here, not a class itself.